Makefile

```diff
@@ -152,10 +155,17 @@ release: all talos-raw talos-gce talos
 test: buildkitd
 	@buildctl --addr $(BUILDKIT_HOST) \
 		build \
-		--exporter=local \
-		--exporter-opt output=./ \
+		--exporter=docker \
+		--exporter-opt output=build/$@.tar \
+		--exporter-opt name=docker.io/autonomy/$@:$(TAG) \
+		--exporter-opt push=$(PUSH) \
 		--frontend-opt target=$@ \
 		$(COMMON_ARGS)
+	@docker load < build/$@.tar
+	@docker run -i --rm $(DOCKER_TEST_ARGS) autonomy/$@:$(TAG) /bin/test.sh --short
+	@trap "rm -rf ./.artifacts" EXIT; mkdir -p ./.artifacts && \
+		docker run -i --rm $(DOCKER_TEST_ARGS) -v $(PWD)/.artifacts:/src/artifacts autonomy/$@:$(TAG) /bin/test.sh && \
+		cp ./.artifacts/coverage.txt coverage.txt
```

Comment:
not sure if we already have .artifacts in .gitignore, but we should make sure it's there

Reply:
it is removed so if it is left, it's a bug...

but on other hand what if you git commit while test is running...

Reply:
fixed